StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetVerticalViewController.swift

```diff
@@ -322,7 +328,15 @@ class PaymentSheetVerticalViewController: UIViewController, FlowControllerViewCo
                 case .link:
                     return isFlowController ? .link : nil // Only default to Link in flow controller mode
                 case .stripeId, nil:
-                    return savedPaymentMethods.first.map { .saved(paymentMethod: $0) }
+                    if let savedSelection = savedPaymentMethods.first {
+                        return .saved(paymentMethod: savedSelection)
+                    }
+                    // If we only have one PM in the list with no wallet options or saved payment methods then select it
+                    if shouldDisplayFormOnly, makeWalletHeaderView() == nil, let paymentMethodType = loadResult.paymentMethodTypes.first {
```

Comment:
It's confusing that we would be in this method when `shouldDisplayFormOnly` is true; that implies we display the form only, not the list.

Reply:
Yeah this usage was a bit overloaded, I've re-written this `if` to check for other more relevant flags to this code path.